rsocket-core/src/main/java/io/rsocket/fragmentation/FrameReassembler.java

```diff
@@ -151,6 +173,19 @@ void cancelAssemble(int streamId) {
   void handleNoFollowsFlag(ByteBuf frame, SynchronousSink<ByteBuf> sink, int streamId) {
     ByteBuf header = removeHeader(streamId);
     if (header != null) {
+
+      int maxReassemblySize = this.maxReassemblySize;
+      if (maxReassemblySize != Integer.MAX_VALUE) {
+        int currentPayloadSize = getMetadataSize(streamId) + getDataSize(streamId);
+        if (currentPayloadSize + frame.readableBytes() - FrameHeaderCodec.size()
+            > maxReassemblySize) {
+          header.release();
+          frame.release();
+          cancelAssemble(streamId);
+          throw new IllegalStateException("Reassembled payload went out of allowed size");
+        }
+      }
+
```

Comment:
If the limit is reached on the very last fragment I think it should be allowed through. It is in memory at this point anyway and the purpose of the limit should be more to prevent further growth after the limit is exceeded or otherwise there would be cases where it fail for even 1 byte over. It should be lenient about that.

Reply:
well. Disagree. This will open another spot of the attack. Since now anyone can flash anything in the pipe, it can configure fragmentation to send the very last frame == MAX FRAME SIZE (16MB) so if the limit of in-memory storage is 16KB, then the last frame will do 16MB + 16KB. Which means - one can use it as a vulnerability